src/maxdiffusion/models/train.py

```diff
@@ -431,21 +432,25 @@ def compute_loss(unet_params):
     max_logging.log(f"number parameters: {num_model_parameters/10**9:.3f} billion")
 
     my_data_sharding = {'input_ids': data_sharding, 'moments': data_sharding}
+    dummy_batch = get_shaped_batch(config, pipeline)
     if (not config.enable_profiler):
-        with jax.transfer_guard("disallow"):
-            p_train_step = jax.jit(
+        with mesh, nn_partitioning.axis_rules(config.logical_axis_rules):
```

Comment:
if you want to remove the transfer_gaurd, I believe you don't need to do this if statement, if and else does the same thing, no matter if the profiler is on or off.